Language/Generics/Superbounded_types/well_bounded_A01_t04.dart

```diff
@@ -16,10 +16,17 @@ class B1<X extends A<Object?>> {}
 class B2<X extends A<dynamic>> {}
 class B3<X extends A<void>> {}
 class B4<X extends A<Never>> {}
+class B5<X extends A<X>> {}
+
+mixin M<X extends A<X>> on A<X> {}
+extension type ET<X extends A<X>>(A<X> _) implements A<X> {}
```

Comment:
Something like this should suffice:
```dart
extension type E<X extends A<Object?>>(String _) implements Object {}
```
We are just interested in the bound of `X`, so there's no need to fight with the representation type or the implemented types, just use something that works.